### ISSUE TITLE
OMEdit layers connection in top instead of bottom layer, and does not remove connectors

### ISSUE THREAD

**Issue by mwetter** [OpenModelica/OpenModelica#11401]
### Description

Consider `Modelica.Fluid.Examples.DrumBoiler.DrumBoiler` shown below.

The connections should be *below* the icons, otherwise the graphic looks strange. See for example `feedback` or `pump`.

Also, the conditionally removed connectors of the `pump` should be removed.

### Screenshots

![image](https://github.com/OpenModelica/OpenModelica/assets/1329080/0d279e8b-33e4-4376-bf8c-af22dab61d44)

### Version and OS
```
OpenModelica 1.23.0~dev-59-g7a41456
OMSimulator 2.1.1~241-g54ebbdc-linux-notlm-debug
```
### Additional Context

This is with "Disable new instance-based graphical editing" selected, which seems to be needed to remove the warnings of https://github.com/OpenModelica/OpenModelica/issues/11388

**Comment by adeas31** [comment]
This is again related to instance-based API. I fixed the issue for old API in #11404 for now. We don't focus on fixing issues related to the old API as it will be removed in the future version.

> Also, the conditionally removed connectors of the pump should be removed.

The conditional connectors are not supported by the old API. Use the new instance-based API.

**Comment by casella** [comment]
@mwetter the old interface is kept as a fallback option for those cases when the new one fails (as it happened during the tutorial in Aachen), but it's not otherwise meant to be used anymore. If there are issues with the new interface, they should be reported and fixed asap.
